lib/extensions/extension_for_provider.dart

```diff
@@ -0,0 +1,23 @@
+import 'package:flutter/cupertino.dart';
+import 'package:provider/provider.dart';
+
+import '../dispose_interface/dispose.dart';
+
+extension WithProvider on Widget {
+  Widget withProvider<T>(T store) => Provider(
```

Comment:
Rename the store variable to the provider as it is not compulsory that we are going to use store only.